ResearchKit/ActiveTasks/ORKTimedWalkStepViewController.m

```diff
@@ -0,0 +1,127 @@
+
+
+#import "ORKTimedWalkStepViewController.h"
+#import "ORKTimedWalkContentView.h"
+#import "ORKTimedWalkStep.h"
+#import "ORKPedometerRecorder.h"
+#import "ORKActiveStepViewController_Internal.h"
+#import "ORKNavigationContainerView_Internal.h"
+#import "ORKStepViewController_Internal.h"
+#import "ORKActiveStepView.h"
+
+
+double const kDistanceTrackingThreshold = 100.0;
+
+
+@interface ORKTimedWalkStepViewController ()
+
+@property (nonatomic, strong) NSMutableArray *samples;
+@property (nonatomic, strong) ORKTimedWalkContentView *timedWalkContentView;
+@property (nonatomic, assign) NSTimeInterval trialDuration;
+
+@end
+
+@implementation ORKTimedWalkStepViewController
+
+- (instancetype)initWithStep:(ORKStep *)step {
+    self = [super initWithStep:step];
+    
+    if (self) {
+        self.suspendIfInactive = YES;
+    }
+    
+    return self;
+}
+
+- (ORKTimedWalkStep *)timedWalkStep {
+    return (ORKTimedWalkStep *)self.step;
+}
+
+- (void)initializeInternalButtonItems {
+    [super initializeInternalButtonItems];
+    
+    self.internalDoneButtonItem = nil;
+    self.continueButtonTitle = ORKLocalizedString(@"BUTTON_DONE", nil);
+}
+
+- (void)viewDidLoad {
+    [super viewDidLoad];
+    
+    self.timedWalkContentView = [ORKTimedWalkContentView new];
+    self.timedWalkContentView.image = [self timedWalkStep].image;
+    self.activeStepView.activeCustomView = self.timedWalkContentView;
+    self.activeStepView.stepViewFillsAvailableSpace = YES;
+    self.activeStepView.continueSkipContainer.continueEnabled = YES;
+    
+    self.timerUpdateInterval = 0.1f;
+}
+
+- (void)finish {
+    [super finish];
+    
+    [self goForward];
+}
+
+- (void)countDownTimerFired:(ORKActiveStepTimer *)timer finished:(BOOL)finished {
+    self.trialDuration = timer.runtime;
+    [super countDownTimerFired:timer finished:finished];
+}
+
+- (ORKStepResult *)result {
+    ORKStepResult *sResult = [super result];
+    
+    NSMutableArray *results = [NSMutableArray arrayWithArray:sResult.results];
+    
+    ORKTimedWalkResult *timedWalkResult = [[ORKTimedWalkResult alloc] initWithIdentifier:(NSString *__nonnull)self.step.identifier];
+    timedWalkResult.distance = [self timedWalkStep].distance;
+    timedWalkResult.timeLimit = [self timedWalkStep].stepDuration;
+    timedWalkResult.duration = self.trialDuration;
+    
+    [results addObject:timedWalkResult];
+    
+    sResult.results = [results copy];
+    
+    return sResult;
+}
+
+#pragma mark - ORKPedometerRecorderDelegate
+
+- (void)pedometerRecorderDidUpdate:(ORKPedometerRecorder *)pedometerRecorder {
+    double timedWalkDistance = [self timedWalkStep].distance;
+    if (timedWalkDistance >= kDistanceTrackingThreshold) {
+        double distanceInMeters = timedWalkDistance - pedometerRecorder.totalDistance;
+        BOOL isValid = pedometerRecorder.totalDistance > 0;
+        [(ORKTimedWalkContentView *)self.activeStepView.activeCustomView setDistanceInMeters:distanceInMeters visible:isValid];
```

Comment:
We have some concerns over the distance being read from pedometer. Due to the participants of this test could have gait and may use assistance device,  which can make pedometer reading inaccurate. Is this displayed inaccurate distance helpful to the participants? 

From another angle, I believe the test require user to put the phone in the pocket, then this label cannot be seen during  the test.